week3/homework/src/actions/markAsDone.js

```diff
@@ -0,0 +1,23 @@
+async function markAsDone(todo, req, res) {
+  try {
+    const todoList = await todo.readList().then(todoList => JSON.parse(todoList));
+    const index = parseInt(req.params.id);
+    if (index > 0 && index <= todoList.length) {

```

Comment:
👍  good job on catching Index out of bounds. Here is a small tip for the future. To avoid nesting if statements like you have done, you can use a so called *early return*
```
if ( index < 0 || index >= todoList.length ) {
   res.status(404).send({ Failed: 'Please insert a valid id'});
   return;
}
// notice how this is no longer nested
if ( todoList ...
```